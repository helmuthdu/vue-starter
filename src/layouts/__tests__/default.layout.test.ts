import { mount, VueWrapper } from '@vue/test-utils';
import { createMemoryHistory, createRouter } from 'vue-router';
import DefaultLayout from '../default.layout.vue';

const router = createRouter({ history: createMemoryHistory(), routes: [] });

describe('Layout -> default', () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(DefaultLayout, { global: { plugins: [router] } });
  });

  it('should renders with props', () => {
    expect(wrapper.exists()).toBe(true);
  });
});
