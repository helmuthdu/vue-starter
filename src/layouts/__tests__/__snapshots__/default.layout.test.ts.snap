// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout -> default should renders with props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <!---->
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "RouterLink": Object {
          "name": "RouterLink",
          "props": Object {
            "activeClass": [Function],
            "ariaCurrentValue": Object {
              "default": "page",
              "type": [Function],
            },
            "custom": [Function],
            "exactActiveClass": [Function],
            "replace": [Function],
            "to": Object {
              "required": true,
              "type": Array [
                [Function],
                [Function],
              ],
            },
          },
          "setup": [Function],
        },
        "RouterView": Object {
          "__emits": Object {},
          "__props": Array [
            Object {
              "name": Object {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
              "route": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [
              "name",
            ],
          ],
          "inheritAttrs": false,
          "name": "RouterView",
          "props": Object {
            "name": Object {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
            "route": [Function],
          },
          "setup": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {
          "$router": Object {
            "addRoute": [Function],
            "afterEach": [Function],
            "back": [Function],
            "beforeEach": [Function],
            "beforeResolve": [Function],
            "currentRoute": RefImpl {
              "__v_isRef": true,
              "_rawValue": Object {
                "fullPath": "/",
                "hash": "",
                "matched": Array [],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
              "_shallow": true,
              "_value": Object {
                "fullPath": "/",
                "hash": "",
                "matched": Array [],
                "meta": Object {},
                "name": undefined,
                "params": Object {},
                "path": "/",
                "query": Object {},
                "redirectedFrom": undefined,
              },
            },
            "forward": [Function],
            "getRoutes": [Function],
            "go": [Function],
            "hasRoute": [Function],
            "install": [Function],
            "isReady": [Function],
            "onError": [Function],
            "options": Object {
              "history": Object {
                "base": "",
                "createHref": [Function],
                "destroy": [Function],
                "go": [Function],
                "listen": [Function],
                "location": "",
                "push": [Function],
                "replace": [Function],
                "state": Object {},
              },
              "routes": Array [],
            },
            "push": [Function],
            "removeRoute": [Function],
            "replace": [Function],
            "resolve": [Function],
          },
        },
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {
        Symbol([vue-router]: router): Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
        Symbol([vue-router]: route location): Object {
          "fullPath": "/",
          "hash": "",
          "matched": Array [],
          "meta": Object {},
          "name": undefined,
          "params": Object {},
          "path": "/",
          "query": Object {},
          "redirectedFrom": undefined,
        },
        Symbol([vue-router]: router view location): RefImpl {
          "__v_isRef": true,
          "_rawValue": Object {
            "fullPath": "/",
            "hash": "",
            "matched": Array [],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
          "_shallow": true,
          "_value": Object {
            "fullPath": "/",
            "hash": "",
            "matched": Array [],
            "meta": Object {},
            "name": undefined,
            "params": Object {},
            "path": "/",
            "query": Object {},
            "redirectedFrom": undefined,
          },
        },
      },
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {
        "$router": Object {
          "addRoute": [Function],
          "afterEach": [Function],
          "back": [Function],
          "beforeEach": [Function],
          "beforeResolve": [Function],
          "currentRoute": RefImpl {
            "__v_isRef": true,
            "_rawValue": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
            "_shallow": true,
            "_value": Object {
              "fullPath": "/",
              "hash": "",
              "matched": Array [],
              "meta": Object {},
              "name": undefined,
              "params": Object {},
              "path": "/",
              "query": Object {},
              "redirectedFrom": undefined,
            },
          },
          "forward": [Function],
          "getRoutes": [Function],
          "go": [Function],
          "hasRoute": [Function],
          "install": [Function],
          "isReady": [Function],
          "onError": [Function],
          "options": Object {
            "history": Object {
              "base": "",
              "createHref": [Function],
              "destroy": [Function],
              "go": [Function],
              "listen": [Function],
              "location": "",
              "push": [Function],
              "replace": [Function],
              "state": Object {},
            },
            "routes": Array [],
          },
          "push": [Function],
          "removeRoute": [Function],
          "replace": [Function],
          "resolve": [Function],
        },
      },
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.7",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
